workflows/src/main/kotlin/com/r3/corda/lib/tokens/workflows/internal/flows/finality/ObserverAwareFinalityFlow.kt

```diff
@@ -68,10 +83,42 @@ class ObserverAwareFinalityFlow private constructor(
         }
         // Sign and finalise the transaction, obtaining the signing keys required from the LedgerTransaction.
         val ourSigningKeys = ledgerTransaction.ourSigningKeys(serviceHub)
-        val stx = transactionBuilder?.let {
-            serviceHub.signInitialTransaction(it, signingPubKeys = ourSigningKeys)
-        } ?: signedTransaction
-        ?: throw IllegalArgumentException("Didn't provide transactionBuilder nor signedTransaction to the flow.")
+
+        val stx = if (haltForExternalSigning) {
+            await(SignTransactionOperation(transactionBuilder!!, ourSigningKeys, serviceHub))
```

Comment:
Remove !! because it throws null pointer exception without explaining what happened. Have meaningful error handling if the transaction builder is null